storage/statedb/database.go

```diff
@@ -311,38 +311,31 @@ func expandNode(hash hashNode, n node) node {
 // NewDatabase creates a new trie database to store ephemeral trie content before
 // its written out to disk or garbage collected.
 func NewDatabase(diskDB database.DBManager) *Database {
-	return NewDatabaseWithNewCache(diskDB, 0)
+	return NewDatabaseWithNewCache(diskDB, GetEmptyTrieNodeCacheConfig())
 }
 
 // NewDatabaseWithNewCache creates a new trie database to store ephemeral trie content
 // before its written out to disk or garbage collected. It also acts as a read cache
 // for nodes loaded from disk.
-func NewDatabaseWithNewCache(diskDB database.DBManager, cacheSizeMB int) *Database {
-	var trieNodeCache Cache
-	var cacheSizeByte int
-
-	if cacheSizeMB == AutoScaling {
-		cacheSizeMB = getTrieNodeCacheSizeMB()
+func NewDatabaseWithNewCache(diskDB database.DBManager, cacheConfig TrieNodeCacheConfig) *Database {
+	trieNodeCache, err := NewTrieNodeCache(cacheConfig)
+	if err != nil {
+		logger.Error("Invalid trie node cache config", "err", err, "config", cacheConfig)
 	}
-	if cacheSizeMB > 0 {
-		cacheSizeByte = cacheSizeMB * 1024 * 1024
-		trieNodeCache = NewFastCache(cacheSizeByte)
 
-		logger.Info("Initialize trie node cache with fastcache", "MaxMB", cacheSizeMB)
-	}
 	return &Database{
-		diskDB:             diskDB,
-		nodes:              map[common.Hash]*cachedNode{{}: {}},
-		preimages:          make(map[common.Hash][]byte),
-		trieNodeCache:      trieNodeCache,
-		trieNodeCacheLimit: cacheSizeByte,
+		diskDB:                  diskDB,
+		nodes:                   map[common.Hash]*cachedNode{{}: {}},
+		preimages:               make(map[common.Hash][]byte),
+		trieNodeCache:           trieNodeCache,
+		trieNodeLocalCacheLimit: cacheConfig.FastCacheSizeMB,
```

Comment:
What do you think of removing this settting and defining `GetSize()` in `TrieNodeCache interface`?

Reply:
Uhm... Can I consider this later? Currently, the value is only used for cache warming up, however we consider more thing such as getting size limit of hybrid cache and redis cache.